### PR TITLE
Fix jenkins dependencies check

### PR DESCRIPTION
**Pull request by jotadrilo**
Currently, we are generating two different locked plugins list for both, the project file and the war file.

However, this is not enough to compare optional dependencies in the project file that are dependencies in the war file if you don't want to add optional dependencies by default.

I am simplifying the approach and create an initial merge in order to take all the required plugins into account.

### CODE REVIEW

File: cmd/jpresolver/main.go [hunk 1]
```diff
@@ -34,9 +34,71 @@ var (
...
 
-func lockPlugins(pr *api.PluginsRegistry, input string) (*api.PluginsRegistry, error) {
+// mergePlugins returns the merge of two slices of plugins with some caveats:
+//
+// The requested plugins are provided by the user and the bundled plugins are Jenkins dependencies.
+// Therefore, we will include all the requested versions that meet the Jenkins requirements. Otherwise,
+// it will fail.
+//
+// Any additional bundled plugin (not present in the requested plugin list) will be included too.
+func mergePlugins(requestedPlugins []*api.Plugin, bundledPlugins []*api.Plugin) ([]*api.Plugin, error) {
```
Comment (atomatt) [inline cmd/jpresolver/main.go]: I think the core logic would be simpler if `plugins` was `pluginsByName := map[string]*api.Plugin{}` and you created a similar `bundledPluginsByName := map[string]*api.Plugin{}` from `bundledPlugins`. It *should* remove a couple of levels of indentation ... I think.

(It's also potentially more scalable as the number of plugins grows, but I doubt that really here.) 

Up to you.

File: cmd/jpresolver/main.go [hunk 2]
```diff
@@ -34,9 +34,71 @@ var (
...
+	var errs error
```
Comment (atomatt) [inline cmd/jpresolver/main.go]: Strictly speaking `errs` is always a singular error; it just happens to be a `multierror`.

Comment (jotadrilo) [reply cmd/jpresolver/main.go]: you are right, but I think I prefer to maintain the convention.

File: cmd/jpresolver/main.go (hunk 1 above)
Comment (jotadrilo) [reply cmd/jpresolver/main.go]: I could refactor how I deal with that list but I am not doing that in this PR. However, I like the idea of indexing the plugins. An indexed map is easier to navigate and work with.

File: cmd/jpresolver/main.go [hunk 3]
```diff
@@ -28,15 +29,79 @@ const (
...
+				}
+			}
+			return nil
+		}
+
+		if err := compatible(requestedPlugin); err != nil {
+			errs = multierror.Append(errs, errors.Trace(err))
+			continue
+		}
+		requestedPlugin.Requester = requesters.PROJECT
```
Comment (atomatt) [inline cmd/jpresolver/main.go]: I wonder if it makes more sense to set `Requester` at the time the plugins are scanned?

**Pull request merged**